### ISSUE TITLE
git lfs migrate export way too long without userfeedback

### ISSUE THREAD

**Issue by axkibe** [git-lfs/git-lfs#5237]
**Describe the issue**
I'm running ```git lfs migrate export --include="*" --everything```
on a very large repository.

First the process does a lot of CPU (without any userfeedback what it is doing)

then it seemd to hang a long way on following syscall (acording to strace)
futex(0xe488a8, FUTEX_WAIT_PRIVATE, 0, NULL

Now it resumed after many minutes.. (i guess, because it's multithreaded).

Could you maybe give the user some head ups that something is happening and what is happening?

BTW: Is this the correct way to leave LFS for a repository (don't worry, I'm staying with LFS but I want this repository LFS-less for testing/developing purposes).

I tried just master instead --everything .. but that does nothing, could you add an export everything example to Readme (for maybe only one ref)? I'm pretty confused how export is working.

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.  I would definitely recommend reading through the `git-lfs-migrate(1)` [manual page](https://github.com/git-lfs/git-lfs/blob/4140f638e9e305605f2e8cd941355340bb71fbf0/docs/man/git-lfs-migrate.adoc), if you haven't already; there's a section there specifically about the `git lfs migrate export` sub-command.

At a high level, `git lfs migrate export`, like `git lfs migrate import`, will rewrite your Git history (changing commit SHAs in the process) to remove or negate Git LFS entries in `.gitattributes` files and change any Git LFS objects matching those entries' patterns into regular Git objects.

If you're using Git LFS in a repository and want to "undo" some or all of the rules in your `.gitattributes` files, and convert the corresponding Git LFS objects into regular Git blobs, then yes, `git lfs migrate export` is an effective tool, so in that regard it's a recommended way to stop using Git LFS.  That said, note that just like the `import` mode, the `export` mode will rewrite your Git history, so you'll want to force-push all the branches and tags of your repository to your remote (assuming things look good once the command is finished).

The `--everything` option means "rewrite history on all refs", as opposed to just the current ref.  The `--include="*"` option is going to cause a final `* !text !filter !merge !diff` rule to be added to any `.gitattributes` files, which should override any other patterns which assign specific file extensions or paths to Git LFS.  That should work, but you might also prefer a more targeted set of rules, in which case providing one or more specific patterns like `--include="*.jpg"` will work.

As for providing some progress information, you can use `GIT_TRACE=1 git lfs migrate export ...` to get a more complete trace of what's being done.  It would be a potential enhancement for the command to provide more intermediate feedback by default, though, so I'll mark this issue as an enhancement request.

Also, note that if Git LFS has to fetch object contents from your remote, that may also take a while (since you may not have all objects stored locally in `.git/lfs`, so the `export` subcommand may have to fetch them from the server in order to convert them into regular Git blobs).

**Comment by axkibe** [comment]
Yes thank you for your detailed explanation, next to the suggestion to give some immediate user feedback what is happening, I suggest for the README to give "git lfs migrate export --include="*"" as a concrete example at the end (with optionally --everything) since IMO this is what most users want to use export for (filtering specific files again is sure useful in some occasions, but IMO a more specialized situation).

**Comment by axkibe** [comment]
PS: in my case, I just wanted a local repository without LFS as testing stage, no need to ever push to that remote again (force or not), but yes it occured to me removing the remote before export is not a good idea ;)

**Comment by chrisd8088** [comment]
The `README.md` file already [contains](https://github.com/git-lfs/git-lfs/blob/4140f638e9e305605f2e8cd941355340bb71fbf0/README.md#uninstalling) an example of how to use `git lfs migrate export`.  It intentionally doesn't use `--include="*"` because the example is designed to align with the `import` example and therefore shows a specific file extension.

I'm inclined to leave it like that because, while I think it's possible a few people will feel comfortable using a blanket `--include="*"`, using specific file extensions is probably what we would recommend in most cases.  Among other things, the subcommand is explicitly designed not to run with `--include="*"` as a silent default when no `--include` or `--exclude` is provided, unlike the `info` and `import` subcommands.

**Comment by axkibe** [comment]
Sorry my fault, I wrote README but was thinking of the manpage:

https://github.com/git-lfs/git-lfs/blob/main/docs/man/git-lfs-migrate.adoc

It only has import examples, no export examples at all.

**Comment by chrisd8088** [comment]
Ah, that makes more sense.  I've opened #5239 to add some examples to the manual page; given others' holiday plans, it may not be merged until next year.

**Comment by axkibe** [comment]
Thanks looks great! This is no urgency anyway.
